ballerina/tests/tests.bal

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/ai;
+import ballerina/test;
+
+const SERVICE_URL = "http://localhost:8080/llm/azureopenai";
+const DEPLOYMENT_ID = "gpt4onew";
+const API_VERSION = "2023-08-01-preview";
+const API_KEY = "not-a-real-api-key";
+const ERROR_MESSAGE = "Error occurred while attempting to parse the response from the LLM as the expected type. Retrying and/or validating the prompt could fix the response.";
+
+final OpenAiProvider openAiProvider = check new (SERVICE_URL, API_KEY, DEPLOYMENT_ID, API_VERSION);
+
+@test:Config
+function testGenerateFunctionWithBasicReturnType() returns ai:Error? {
+    int rating = check openAiProvider.generate(`Rate this blog out of 10.
+        Title: ${blog1.title}
+        Content: ${blog1.content}`);
+    test:assertEquals(rating, 4);
+}
+
+@test:Config
+function testGenerateFunctionWithRecordReturnType() returns error? {
+    Review result = check openAiProvider.generate(`Please rate this blog out of 10.
+        Title: ${blog2.title}
+        Content: ${blog2.content}`);
+    test:assertEquals(result, check review.fromJsonStringWithType(Review));
+}
+
+@test:Config
+function testGenerateFunctionWithBasicReturnTypeWithTextDocument() returns ai:Error? {
```

Comment:
Aren't we testing the text document part here? How does `BasicReturnType` matter?

Reply:
Let's carefully plan and add the necessary and sufficient tests.